### PR TITLE
Add a usage guide for the scripts in the readme

### PR DESCRIPTION
**Pull request by Madman10K**
Explains the following scripts and how they should be used:
1. `process.py`
1. `run.sh`
1. `personal-run.sh`

### CODE REVIEW

**Comment by Madman10K** [comment]
Rename `personal-run.sh` to `youyin-dev-run.sh` because it makes more sense

File: README.md [hunk 1]
```diff
@@ -21,3 +23,30 @@ How to use:
...
+Next, edit `YouyinWeb/index.js` and change the `window.CHARACTER_FETCH_URL` string to `"http://0.0.0.0:8080/data/"`
+
+With this setup, you can test characters. Now, every time you have finished a batch of characters, go into the `make-me-a-hanzi-tool/server` directory and run `./youyin-dev-run.sh`. It will:
+
+1. Create `hanzi-writer` character files, like `run.sh`
+1. Copy them to `hanzi-writer-data-youyin/data`
+1. Go into `YouyinWeb` and run `./run.sh` to start the server.
+1. Finally, open Youyin locally on `http://0.0.0.0:8080` and try creating character cards with the new characters.
+
+Once all characters are working and generated correctly, simply got to the `hanzi-writer-data-youyin` directory, commit and push the changes.
```
Comment (nick-bors) [inline README.md]: Typo
```suggestion
Once all characters are working and generated correctly, simply go to the `hanzi-writer-data-youyin` directory, commit and push the changes.
```

File: README.md [hunk 2]
```diff
@@ -21,3 +23,30 @@ How to use:
...
+Create a symbolic link so that `hanzi-writer-data-youyin/data/` points to `YouyinWeb/data/`: `ln -s YouyinWeb/data/ hanzi-writer-data-youyin/data/`
+
+Next, edit `YouyinWeb/index.js` and change the `window.CHARACTER_FETCH_URL` string to `"http://0.0.0.0:8080/data/"`
+
+With this setup, you can test characters. Now, every time you have finished a batch of characters, go into the `make-me-a-hanzi-tool/server` directory and run `./youyin-dev-run.sh`. It will:
+
+1. Create `hanzi-writer` character files, like `run.sh`
+1. Copy them to `hanzi-writer-data-youyin/data`
+1. Go into `YouyinWeb` and run `./run.sh` to start the server.
+1. Finally, open Youyin locally on `http://0.0.0.0:8080` and try creating character cards with the new characters.
```
Comment (fami-fish) [inline README.md:50]: Numbering
```suggestion
1. Create `hanzi-writer` character files, like `run.sh`
2. Copy them to `hanzi-writer-data-youyin/data`
3. Go into `YouyinWeb` and run `./run.sh` to start the server.
4. Finally, open Youyin locally on `http://0.0.0.0:8080` and try creating character cards with the new characters.
```

File: README.md [hunk 3]
```diff
@@ -21,3 +23,30 @@ How to use:
...
+
+1. `process.py` - Reads the `make-me-a-hanzi` dictionary from the characters you have created and converts it to individual character files in `hanzi-writer` format under the `output` folder. Do not use directly, use `run.sh` instead
+1. `run.sh` - Gets the character data from the meteor database, exports it to `tmp.json`, which is used as a `make-me-a-hanzi` dictionary file. It then runs `process.py`
+
+### `youyin-dev-run.sh`
+This script runs `./run.sh` for Youyin development. First, make sure you have the following repositories cloned in the same folder without renames:
+
+1. make-me-a-hanzi-tool
+1. [YouyinWeb](https://github.com/MadLadSquad/YouyinWeb)
+1. [hanzi-writer-data-youyin](https://github.com/MadLadSquad/hanzi-writer-data-youyin)
```
Comment (fami-fish) [inline README.md:39]: Numbering
```suggestion
1. make-me-a-hanzi-tool
2. [YouyinWeb](https://github.com/MadLadSquad/YouyinWeb)
3. [hanzi-writer-data-youyin](https://github.com/MadLadSquad/hanzi-writer-data-youyin)
```

File: README.md (hunk 2 above)
Comment (nick-bors) [inline README.md]: Unordered list makes more sense here
```suggestion
- Create `hanzi-writer` character files, like `run.sh`
- Copy them to `hanzi-writer-data-youyin/data`
- Go into `YouyinWeb` and run `./run.sh` to start the server.
- Finally, open Youyin locally on `http://0.0.0.0:8080` and try creating character cards with the new characters.
```

File: README.md (hunk 3 above)
Comment (nick-bors) [inline README.md]: Unordered list
```suggestion
- make-me-a-hanzi-tool
- [YouyinWeb](https://github.com/MadLadSquad/YouyinWeb)
- [hanzi-writer-data-youyin](https://github.com/MadLadSquad/hanzi-writer-data-youyin)
```

File: README.md [hunk 4]
```diff
@@ -21,3 +23,30 @@ How to use:
     3. R - Clear changes in the stroke editor
     4. N - Next character
 6. Note: When cutting 2 intersecting strokes, create 1 or 2 rectangles in the overlapped area when cutting. If any unneeded strokes appear deselect them
+
+## Additional script usage for Youyin
+In addition to improving the editor, we've also added scripts to make generating data for [youyin](https://github.com/MadLadSquad/YouyinWeb) and 
+[hanzi-writer](https://github.com/chanind/hanzi-writer) easier. All scripts are located under the `server/` directory because that's where the character data is generated.
+
+1. `process.py` - Reads the `make-me-a-hanzi` dictionary from the characters you have created and converts it to individual character files in `hanzi-writer` format under the `output` folder. Do not use directly, use `run.sh` instead
+1. `run.sh` - Gets the character data from the meteor database, exports it to `tmp.json`, which is used as a `make-me-a-hanzi` dictionary file. It then runs `process.py`
```
Comment (nick-bors) [inline README.md]: UL
```suggestion
- `process.py` - Reads the `make-me-a-hanzi` dictionary from the characters you have created and converts it to individual character files in `hanzi-writer` format under the `output` folder. Do not use directly, use `run.sh` instead
- `run.sh` - Gets the character data from the meteor database, exports it to `tmp.json`, which is used as a `make-me-a-hanzi` dictionary file. It then runs `process.py`
```

File: README.md [hunk 5]
```diff
@@ -21,3 +23,30 @@ How to use:
     3. R - Clear changes in the stroke editor
     4. N - Next character
```
Comment (nick-bors) [inline README.md:24]: Missing alot of bindings. Also, i never heard of `N` before

File: README.md [hunk 6]
```diff
@@ -1,7 +1,8 @@
 # make-me-a-hanzi-tool
 A (probably) working version of the make-me-a-hanzi tool.
 
-How to build:
+Building:
+
 1. Install mongodb
 1. Install node and run `npm i`
 1. Install meteor using the following command:
```
Comment (nick-bors) [inline README.md:8]:
```suggestion
1. Install mongodb
2. Install node and run `npm i`
3. Install meteor using the following command:
```

**Review by nick-bors — CHANGES_REQUESTED** [review]
few errors lol

File: README.md (hunk 5 above)
Comment (Madman10K) [reply README.md:24]: Well, `N` could be one of the following:
1. `n`
1. `j`
1. `k`

Colemak user moment :skull:

**Pull request merged**